eth/handler.go

```diff
@@ -413,13 +436,21 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 			}
 		}
 		if len(headers) == 1 {
-			if err := pm.chainConfig.HeaderCheck(headers[0]); err != nil {
+			header := headers[0]
+			err := pm.chainConfig.HeaderCheck(header)
+			if err != nil {
 				pm.removePeer(p.id)
 				return err
 			}
+			if !header.Hash().IsEmpty() {
+				p.MarkHeader(header.Hash())
+			}
 			// Irrelevant of the fork checks, send the header to the fetcher just in case
 			headers = pm.fetcher.FilterHeaders(headers, time.Now())
 		}
+		if err := pm.requestRequiredHeaders(p); err != nil {
```

Comment:
Is it a good idea to do fork check every time we receive a new BlockHeaders message? I think fork check should only happen when a new peer is connected.